SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/Utils/DataTypeUtilities.cpp

```diff
@@ -28,8 +28,10 @@ FString GetEnumDataType(const UEnumProperty* EnumProperty)
 
 FString UnrealNameToSchemaTypeName(const FString& UnrealName)
 {
-	// Note: Removing underscores and spaces to avoid naming mismatch between how schema compiler and interop generator process schema identifiers.
-	return UnrealName.Replace(TEXT("_"), TEXT("")).Replace(TEXT(" "), TEXT(""));
+	// Note: Removing underscores, spaces, and capitalizing first letter to avoid naming issues to satisfy schema compiler.
+	FString SchemaTypeName = UnrealName.Replace(TEXT("_"), TEXT("")).Replace(TEXT(" "), TEXT(""));
+	SchemaTypeName[0] = FChar::ToUpper(SchemaTypeName[0]);
```

Comment:
This is now going to capitalize schema field names and command names, which is not allowed. I think we need two functions here, one to remove underscores and spaces, and another one to capitalize.

Reply:
Yep